### PR TITLE
Updated provider to use Upjet instead of Terrajet. Updated Dockerfile to download release from Github instead of releases.hashicorp.com

### PR DESCRIPTION
**Pull request by prakashmishra1598**
### Description of your changes

Updated provider to use Upjet instead of Terrajet. Updated Dockerfile to download release from Github instead of releases.hashicorp.com

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested

Code tested and built locally. All builds were successful.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,18 +1,32 @@
-# Terrajet Palette Provider
+# Provider JetPalette
```
Comment (TylerGillson) [inline README.md:1]: Should be "Palette"

File: examples/providerconfig/providerconfig.yaml [hunk 2]
```diff
@@ -1,11 +1,10 @@
-apiVersion: jet-palette.upbound.io/v1beta1
 kind: ProviderConfig
 metadata:
   name: default
 spec:
   credentials:
     source: Secret
     secretRef:
-      name: example-creds
+      name: pallete-creds
```
Comment (TylerGillson) [inline examples/providerconfig/providerconfig.yaml:8]: typo here :)

File: examples/providerconfig/secret.yaml.tmpl [hunk 3]
```diff
@@ -1,12 +1,13 @@
 apiVersion: v1
 kind: Secret
 metadata:
-  name: example-creds
+  name: pallete-creds
```
Comment (TylerGillson) [inline examples/providerconfig/secret.yaml.tmpl:4]: spelling

File: README.md [hunk 4]
```diff
@@ -1,18 +1,32 @@
...
 
 `provider-jet-palette` is a [Crossplane](https://crossplane.io/) provider that
-is built using [Terrajet](https://github.com/crossplane/terrajet) code
-generation tools and exposes XRM-conformant managed resources for the 
-Palette API.
+is built using [Upjet](https://github.com/upbound/upjet) code
+generation tools and exposes XRM-conformant managed resources for the
+JetPalette API.
```
[the remark this answers is not in the document]
Comment (prakashmishra1598) [reply README.md]: Updated. Please take a look :)

File: examples/null/resource.yaml [hunk 5]
```diff
@@ -1,4 +1,4 @@
-apiVersion: null.palette.jet.crossplane.io/v1alpha1
+apiVersion: null.jet-palette.upbound.io/v1alpha1
```
[the remark this answers is not in the document]
Comment (prakashmishra1598) [reply examples/null/resource.yaml]: Added examples for `cluster` and `cloudaccount`

File: README.md (hunk 1 above)
Comment (prakashmishra1598) [reply README.md]: Updated the PR. PTAL :)

File: examples/providerconfig/providerconfig.yaml (hunk 2 above)
Comment (prakashmishra1598) [reply examples/providerconfig/providerconfig.yaml]: Fixed :)

**Review by TylerGillson — APPROVED** [review]
LGTM

**Review by saamalik — APPROVED** [review]

**Pull request merged**